py/stream.c

```diff
@@ -182,20 +224,27 @@ STATIC mp_obj_t stream_read(size_t n_args, const mp_obj_t *args) {
         return mp_obj_new_str_from_vstr(STREAM_CONTENT_TYPE(stream_p), &vstr);
     }
 }
+
+STATIC mp_obj_t stream_read(size_t n_args, const mp_obj_t *args) {
+    return stream_read_generic(n_args, args, OP_READ);
+}
```

Comment:
I checked that couple of years ago, and now comfortable about that. Ideally, it should be rechecked regarding the best placement of additional args - note that it would be different for stack and register calling conventions. This order should be the best for reg-based CC, as most embedded targets use.